types.tsx

```diff
@@ -24,36 +24,20 @@ declare global {
  */
 export type RootStackParamList = {
   Root: NavigatorScreenParams<RootTabParamList> | undefined;
-  AddVeggieModal: { selectedGardenId: string; selectedBedId: string };
+  AddVeggieModal: { selectedBedId: string };
   NotFound: undefined;
-  NewVeggieLogModal: {
-    selectedGardenId: string;
-    selectedBedId: string;
-    veggieId: string;
-  };
-  EditVeggieLogModal: {
-    selectedGardenId: string;
-    selectedBedId: string;
-    veggieId: string;
-    logId: string;
-  };
-  CreateCardModal: {
-    selectedGardenId?: String;
-    areaTitle: string;
-    routeName: string;
-  };
-  RenameCardModal: {
-    selectedGardenId: string;
-    selectedBedId: string | undefined;
-    routeName: string;
-  };
-  CardOptionsModal: {
-    selectedGardenId: string;
-    selectedBedId?: string;
-    routeName: string;
-    title?: string;
-  };
-  DeleteConfirmationModal: { selectedGardenId: string; selectedBedId?: string };
+  NewVeggieLogModal: { veggieId: string };
+  EditVeggieLogModal: { logId: string };
+  CreateCardModal: { selectedGardenId?: string } | undefined;
+  CardOptionsModal:
```

Comment:
You'll see this around   CardOptionsModal:
     | { selectedGardenId?: never; selectedBedId: string }
     | { selectedGardenId: string; selectedBedId?: never };

it basically means one or the other, not both